### ISSUE TITLE
Should bounded fillRandom() calls change to exclusive upper bounds?

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#20165]
In [this comment chain on Arkouda](https://github.com/Bears-R-Us/arkouda/pull/1542#issuecomment-1176338338) we realized that both Python and PCG use an exclusive upper bound, where Chapel's fillRandom() was defined as being inclusive.  It seems as though changing it to be exclusive would be more consistent / less surprising with what users might expect.  This issue proposes that we change to exclusive while we have separate min/max values.

[It was also suggested that maybe we should take a range to indicate the bounds which would have the advantages of (a) making the inclusive vs. exclusive nature more explicit (i.e., `0..n` vs. `0..<n`), similar to how we're changing channel creation to take its region as a range...  But a challenge in this context is that we don't currently support ranges of floating point values]

**Comment by bradcray** [comment]
Noting that transitioning users from one behavior to the other seems like it will lead to some challenges here...  My thought would be to have a compilerWarning about the change in behavior in a future Chapel release and an opt-in config param to jump to the new behavior now.

**Comment by mppf** [comment]
Or we could migrate to a range at the same time (to trigger deprecation warnings etc). Perhaps there isn't enough code using bounded RNG generation with floating point numbers to worry about. I don't have a sense how much code we have using that.

**Comment by bradcray** [comment]
I agree that we, ourselves, may not have much code that uses floating point RNG, but I think it is a common thing to want (e.g., floating point from [0.0, 1.0).  Of course, we could still introduce a range-taking version for integers and continue to use the min/max values version for floats until a day when we have float ranges.

I like ranges sufficiently that I like that approach and idea, but it should obviously get opinions beyond mine on it.

I forgot to mention above, but you also had concerns about maximal ranges causing problems for bounded RNG, but I felt as though that was less likely to be an issue.  Specifically, I'd imagine `0..<max(myIntType)` to be the common case when things got large (and maybe the only one PCG supports underneath?), but I think as long as it's exclusive like that we'd be OK (?).
